src/service_router.rs

```diff
@@ -0,0 +1,90 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use context::{ ContextTrait, Shared };
+use iron::{Request, Response, IronResult};
+use iron::headers::ContentType;
+use iron::status::Status;
+use router::Router;
+use core::marker::Reflect;
+
+
+pub fn generate<Ctx: Send + Reflect + ContextTrait + 'static> (context: Shared<Ctx>) -> Router {
+
+    let mut router = Router::new();
+    let context1 = context.clone();
+    router.get("list.json", move |_: &mut Request| -> IronResult<Response> {
+        // Build a json representation of the services.
+        let ctx = context1.lock().unwrap();
+        let serialized = itry!(ctx.services_as_json());
+
+        let mut response = Response::with(serialized);
+        response.status = Some(Status::Ok);
+        response.headers.set(ContentType::json());
+
+        Ok(response)
+    });
+
+    let context2 = context.clone();
+    router.get(":service", move |req: &mut Request| -> IronResult<Response> {
```

Comment:
Thanks for pointing this out. I misunderstood the function. The URL I wrote in the tests (like [this one](https://github.com/fxbox/foxbox/pull/48/files#diff-2b028424809a6315630f8281a5f7d1e3R83)) were failing. As I didn't see any `command` being used in this context, I thought it was a nit. 

I fixed the tests by adding a command in the URL.

However, that makes me wonder if we shouldn't make this function owner of `command` and pass it explicitly to the service. For instance:
``` rust
let router_plugin = req.extensions.get::<Router>().unwrap(); // TODO: Find a var name less close to `router`
let id = router_plugin.find("service").unwrap_or("");
let downstream_request = {
    command: router_plugin.find("command").unwrap_or("");
};

match ctx.get_service(id) {
    None => {// ...}
    Some(service) => {
        service.process_request(downstream_request)
    }
}
```

What do you think @fabricedesre ?